src/Agent.Plugins/PipelineArtifact/PipelineArtifactProvider.cs

```diff
@@ -10,11 +10,23 @@
 using Agent.Sdk;
 using System.Threading;
 using System.Linq;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
 
 namespace Agent.Plugins.PipelineArtifact
 {
     internal class PipelineArtifactProvider : IArtifactProvider
     {
+        private const int DefaultDedupStoreClientMaxParallelism = 192;
+
+        internal static int GetDedupStoreClientMaxParallelism(AgentTaskPluginExecutionContext context) {
+            int parallelism = DefaultDedupStoreClientMaxParallelism;
+            if(context.Variables.TryGetValue("AZURE_PIPELINES_DEDUP_PARALLELISM", out VariableValue v)) {
+                parallelism = int.Parse(v.Value);
```

Comment:
Are we certain that the v.Value will always be digits? Should we use Int32.TryParse(v.Value) to be safe? If it's not a number then use default value

Reply:
@TingluoHuang What's the recommendation here? Ignore bad values or fail fast?

Reply:
Ok I followed the pattern I found [here](https://github.com/microsoft/azure-pipelines-agent/blob/d9a8b0774e8b6b9d997c0a77ca9686fe849c4584/src/Agent.Listener/Agent.cs#L186)

```
                    if (!Enum.TryParse(startupTypeAsString, true, out startType))
                    {
                        Trace.Info($"Could not parse the argument value '{startupTypeAsString}' for StartupType. Defaulting to {StartupType.Manual}");
                        startType = StartupType.Manual;
                    }
```